src/classes/RD2_ElevateInformation_CTRL.cls

```diff
@@ -0,0 +1,172 @@
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Displays Elevate recurring commitment information
+*/
+public with sharing class RD2_ElevateInformation_CTRL {
+
+
+    /**
+    * @description Finds records related to Recurring Donation Elevate data
+    */
+    @TestVisible
+    private static RecordFinder finder {
+        get {
+            if (finder == null) {
+                finder = new RecordFinder();
+            }
+            return finder;
+        }
+        set;
+    }
+
+    /**
+    * @description Returns data required for the Elevate Information widget
+    * @param recordId Recurring Donation Id
+    * @return Map<String, Object> Data value by key
+    */
+    @AuraEnabled(cacheable=false)
+    public static Map<String, Object> getData(Id recordId) {
+        Map<String, Object> data = new Map<String, Object>();
+
+        try {
+            data = new Map<String, Object> {
+                'isElevateCustomer' => RD2_ElevateIntegrationService.isElevateCreateEnabled(),
+                'alert' => RD2_VisualizeScheduleController.getValidationAlert(recordId),
+                'hasFieldPermissions' => hasFieldPermissions(),
+                'errorMessage' => getLatestErrorMessage(recordId)
+            };
+
+        } catch (Exception e) {
+            UTIL_AuraEnabledCommon.throwAuraHandledException(e.getMessage());
+        }
+
+        return data;
+    }
+
+    /**
+     * @description Checks if the user has access for fields retrieved by the Elevate information widget
+     * @return Boolean
+     */
+    private static Boolean hasFieldPermissions() {
+        Set<String> requiredFields = new Set<String>{
+            UTIL_Namespace.StrTokenNSPrefix('Status__c'),
+            UTIL_Namespace.StrTokenNSPrefix('ClosedReason__c'),
+            UTIL_Namespace.StrTokenNSPrefix('CommitmentId__c')
+        };
+
+        Boolean hasPermissions = true;
+
+        for (String fld : requiredFields) {
+            DescribeFieldResult dfr = UTIL_Describe.getFieldDescribe('npe03__Recurring_Donation__c', fld);
+            if (!dfr.isAccessible()) {
+                hasPermissions = false;
+                break;
+            }
+        }
+
+        return hasPermissions;
+    }
+
+    /***
+     * @description Displays the latest error when latest payment could not be charged or
+     * any other issue with the recurring commitment in Elevate
+     * @param recordId Recurring Donation Id
+     * @return String The latest error message if no successful payment is created after it.
+     */
+    private static String getLatestErrorMessage(Id recordId) {
```

Comment:
Since FLS check is only done in front end but data is still there in the JS, It will be nice to include the FLS check here so no data will be exposed if the user does not have the correct field permission.

Reply:
That is a great suggestion, I will make the change later today.